### DISCUSSION TITLE
Tiled Diffusion Upscaling (1st Run is Fast, 2nd Run takes Hours)

### DISCUSSION THREAD

**Discussion by gohan2091** [vladmandic/sdnext#2281] [Q&A]
So I am trying to upscale some images using ControlNet Tiles with Tiled Diffusion (Tiled VAE enabled). Actually right now, I have a 2048 x 3072 image that I am leaving the upscaler (UltraSharp 4x) at 1x scale and using a low denoise to try and generate extra details. My issue is, the first time I perform this task in img2img everything is fast. It takes just a few minutes to complete. Once completed, if I leave everything exactly how it was, same image, same settings and just hit the generate button again, everything slows right down. It takes several minutes for the button to go from "starting..." to showing me an estimated time of completion. That time of completion is 300 - 400 minutes  remaining! Why is this? It makes no sense that I am performing the exact same task with same image and setting yet what took me 5 minutes before now takes hundreds of minutes. If I close SD.Next down and reopen it, I can perform the task again in just minutes yet the second time I try it, same issue.... 

Here is a copy and paste of my log while I perform the task. At the top shows me upscaling x1 scale in just minutes, after I attempt to upscale again by x1 the estimate time shows 3 hours:

`Upscaling ---------------------------------------- 100% 0:00:00 0:00:12
[Tiled Diffusion] ControlNet found, support is enabled.
MixtureOfDiffusers Sampling: : 0it [00:00, ?it/s]Mixture of Diffusers hooked into 'DPM++ 3M SDE Karras' sampler, Tile size: 112x112, Tile batches: 2, Batch size: 8. (ext: ContrlNet)
[Tiled VAE]: the input size is tiny and unnecessary to tile.
00:47:30-765749 STATUS   Loaded VAE-approx model: models\VAE-approx\model.pt
Loading weights: C:\Users\USER\Vlads A1111\automatic\models\Lora\LowRA.safetensors --------------- 0.0/75.6 MB -:--:--
Loading weights: C:\Users\USER\Vlads A1111\automatic\models\Lora\vampface.safetensors ------------ 0.0/151.1   -:--:--
                                                                                                    MB
Loading weights: C:\Users\USER\Vlads A1111\automatic\models\Lora\add_detail.safetensors ----------- 0.0/37.9   -:--:--
                                                                                                     MB
Loading weights: C:\Users\USER\Vlads A1111\automatic\models\Lora\detailed_eye-10.safetensors -------- 0.0/19.0 -:--:--
                                                                                                       MB
MixtureOfDiffusers Sampling: : 0it [00:17, ?it/s]
100%|██████████████████████████████████████████████████████████████████████████████████| 55/55 [01:32<00:00,  1.68s/it]
[Tiled VAE]: input_size: torch.Size([1, 4, 384, 256]), tile_size: 192, padding: 11█████| 55/55 [01:32<00:00,  1.65s/it]
[Tiled VAE]: split to 2x2 = 4 tiles. Optimal tile size 128x192, original tile size 192x192
[Tiled VAE]: Fast mode enabled, estimating group norm parameters on 128 x 192 image
[Tiled VAE]: Executing Decoder Task Queue: 100%|█████████████████████████████████████| 492/492 [00:22<00:00, 21.79it/s]
[Tiled VAE]: Done in 26.807s, max VRAM alloc 12285.229 MB
00:49:40-931148 STATUS   Processed: images=1 time=141.08s its=0.39 memory={'ram': {'used': 5.8, 'total': 63.92}, 'gpu':
                         {'used': 8.05, 'total': 23.99}, 'retries': 0, 'oom': 0}
[Tiled Diffusion] upscaling image with 4x-UltraSharp...
00:50:19-556423 STATUS   Model loaded: type=ESRGAN model=C:\Users\USER\Vlads
                         A1111\automatic\models\ESRGAN\4x-UltraSharp.pth
Upscaling ---------------------------------------- 100% 0:00:00 0:00:12
[Tiled Diffusion] ControlNet found, support is enabled.
MixtureOfDiffusers Sampling:   0%|                                                             | 0/110 [00:00<?, ?it/s]Mixture of Diffusers hooked into 'DPM++ 3M SDE Karras' sampler, Tile size: 112x112, Tile batches: 2, Batch size: 8. (ext: ContrlNet)
[Tiled VAE]: the input size is tiny and unnecessary to tile.
MixtureOfDiffusers Sampling:   1%|▍                                                  | 1/110 [01:39<2:59:59, 99.07s/it]
  4%|██▉                                                                             | 2/55 [03:32<1:33:42, 106.08s/it]`

**Comment by vladmandic** [comment]
can you monitor your vram usage in system info?

**Reply by gohan2091** [reply]
Before I start, my 4090 VRAM is around 3.5GB out of 24GB (according to Task Manager). When I am upscaling, my VRAM usage goes to around 14 GB. A few minutes later, the upscale is done. My VRAM doesn't return to 3.5GB, it sits at 7.3GB. I click the "Free GPU" button within Tiled Diffusion which says: 

[Mem] rss: 5.378 GB, vms: 13.254 GB
[VRAM] free: 15.971 GB, total: 23.988 GB

However, Task Manager still shows my VRAM at 7.3GB/24GB. (Changing the model checkpoint to something else then back to the one I want at this stage lowers the VRAM to 4GB) When I go to re-do the upscale with the exact same settings, I notice a few things:

1) Instead of taking a few minutes like before, it now estimates hours
2) The VRAM usage in Task Manager goes to 23/24GB (not the 14GB previously)
3) In the first rescale, the command window says:

MixtureOfDiffusers Sampling: : 0it [00:15, ?it/s]
100%|██████████████████████████████████████████████████████████████████████████████████| 50/50 [01:12<00:00,  1.46s/it]
[Tiled VAE]: input_size: torch.Size([1, 4, 384, 256]), tile_size: 192, padding: 11█████| 50/50 [01:12<00:00,  1.37s/it]
[Tiled VAE]: split to 2x2 = 4 tiles. Optimal tile size 128x192, original tile size 192x192
[Tiled VAE]: Fast mode enabled, estimating group norm parameters on 128 x 192 image
[Tiled VAE]: Executing Decoder Task Queue: 100%|█████████████████████████████████████| 492/492 [00:23<00:00, 20.61it/s]

Notice the number/50 on both lines. In the second rescale, instead of number/50 it says in the first line number/100 and second line number/50. Any clues with that? It should be the same as the first right?

My Tiled VAE settings are: Encoder Tile Size of 3072 and Decoder Tile Size of 192 (default)

**Comment by josh5671** [comment]
From what i guess. you not using 531 nvidia driver right?
also, the tiled vae default 3072,192 actually not so perfect on my 3090 24gb , vram will explode (OOM)

try check your nvidia driver version. and lower the Encoder size.

**Reply by gohan2091** [reply]
I'm using driver 537.42. If the TiledVAE settings are not great, then why does it work the first time fine without any issue? My VRAM is only around half first time.

**Comment by gohan2091** [comment]
Made a fresh install of SD.Next and it's working fine now... I can upscale multiple times and no issue!
